skbio/io/fastq.py

```diff
@@ -447,18 +465,19 @@ def _alignment_to_fastq(obj, fh, variant=None, phred_offset=None,
                         description_newline_replacement)
 
 
-def _line_generator(fh):
-    for line in fh:
-        line = line.rstrip('\n')
-        if not line:
-            raise FASTQFormatError("Found blank line in FASTQ-formatted file.")
-        yield line
+def _blank_error(unique_text):
+    error_string = ("Found blank or whitespace-only line {} in "
+                    "FASTQ file").format(unique_text)
+    raise FASTQFormatError(error_string)
 
 
 def _parse_sequence_data(fh):
     seq_chunks = []
+    prev = '@'
```

Comment:
It might be clearer (and safer, in case the caller's logic changes in the future) to update `_parse_sequence_data` and `_parse_quality_scores` to accept `prev` as a positional argument. Then the callers would supply their actual previous line when calling this function instead of faking a previous line.